csrc/ir/utils.cpp

```diff
@@ -1313,9 +1314,37 @@ bool hasTrivialAllocationDomain(const TensorView* tv) {
   }
   const std::vector<IterDomain*>& alloc = tv->getMaybeAllocationDomain();
   const std::vector<IterDomain*>& logical = tv->getLogicalDomain();
-  return TensorDomain::noBroadcasts(TensorDomain::noReductions(logical)) ==
+  const auto alloc_no_red_bcast =
```

Comment:
I'm unsure about changing hasTrivialAllocationDomain, which is used elsewhere. Philosophically, its name suggests it should check for the allocation being trivial and having a DID split is conceptually non-trivial. 

More practically, how about using inferShapeOfOutput? I made it support DID loop split and it's [used](https://github.com/NVIDIA/Fuser/blob/ac84633b89f26cbccd242bd3d05153c4dfa2b5ea/csrc/runtime/allocations.cpp#L277) by KernelExecutor to allocate outputs. I *think* in MatmulOp::evaluate you can say something like 

```
sizes, strides = inferShapeOfOutput(tv, ee);
create a meta tensor of sizes and strides
if that_meta_tensor.is_contiguous():
  return the result of at::matmul
...
```

When you change the code to use at::matmul_out (which I still think is a change we should land sooner than later), you can further simplify MatmulOp::evaluate to:

```
sizes, strides = inferShapeOfOutput(tv, ee);
out = at::empty_strided(sizes, strides, ...);
at::matmul_out(...);
```